### PR TITLE
remove deepcopy from grad! and obj!

### PR DESCRIPTION
**Pull request by grishabh147**
This PR brings the following two changes:
- Removes unnecessary code from the test directory
- updates `grad!` and `obj!` implementations for augmented nlp blocks.

### CODE REVIEW

File: src/augmented_block.jl [hunk 1]
```diff
@@ -57,7 +58,7 @@ function AugmentedNLPBlockModel(
         (block_hess_struct[1], block_hess_struct[2]),
         ATA,
     )
-    return AugmentedNLPBlockModel(meta, Counters(), nlp, λ, ρ, A, b, sol, hess_struct)
+    return AugmentedNLPBlockModel(meta, Counters(), nlp, λ, ρ, A, b, sol, sol, hess_struct)
```
Comment (frapac) [inline src/augmented_block.jl]: why do we need two `sol` here?

**Review by frapac — APPROVED** [review]
Nice catch!

File: src/augmented_block.jl (hunk 1 above)
Comment (grishabh147) [reply src/augmented_block.jl:61]: The second one is to initialize `local_sol` with the current solution, which gets updated as every block is solved in an iteration of some decomposition algorithm.
 
The first `sol` gets updated only once all the blocks have been solved.

Comment (grishabh147) [reply src/augmented_block.jl]: Yeah, you are right. Thanks for pointing this out.

We do not need two `sol` for `AugmentedNLPBlockModel`. I have updated the code.

We do need an additional vector for the `ProxAugmentedNLPBlockModel` to be able to compute the proximal term.

**Pull request merged**